File: uxmobile/src/main/java/sk/brecka/uxmobile/model/event/ExceptionEvent.java
package sk.brecka.uxmobile.model.event;

import org.json.JSONArray;
import org.json.JSONException;

import java.io.PrintWriter;
import java.io.StringWriter;

import sk.brecka.uxmobile.model.event.Event;

/**
 * Created by matej on 7.2.2018.
 */

public class ExceptionEvent extends Event {
    protected static final int INDEX_STACKTRACE = 2;

    private final Throwable mThrowable;

    public ExceptionEvent(long startTime, Throwable throwable) {
        super(startTime);
        mThrowable = throwable;
    }

    @Override
    protected String getType() {
        return TYPE_EXCEPTION;
    }

    @Override
    public JSONArray toJson() throws JSONException {

        StringWriter sw = new StringWriter();
        PrintWriter pw = new PrintWriter(sw);
        mThrowable.printStackTrace(pw);
        String stackTrace = sw.toString();

        return super.toJson()
                .put(INDEX_STACKTRACE, stackTrace);
    }
}

File: uxmobile/src/main/java/sk/uxtweak/uxmobile/model/event/ExceptionEvent.kt
package sk.uxtweak.uxmobile.model.event

import org.json.JSONArray
import java.io.PrintWriter
import java.io.StringWriter

class ExceptionEvent(startTime: Long, private val throwable: Throwable) : Event(startTime) {
    override fun getType() = TYPE_EXCEPTION

    override fun toJson(): JSONArray {
        val stringWriter = StringWriter()
        val printWriter = PrintWriter(stringWriter)
        throwable.printStackTrace(printWriter)
        val stackTrace = stringWriter.toString()

        return super.toJson().put(INDEX_STACKTRACE, stackTrace)
    }

    companion object {
        const val INDEX_STACKTRACE = 2
    }
}
